// dolphinng/components/header/header.component.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var HeaderComponent = (function () {
    function HeaderComponent() {
    }
    return HeaderComponent;
}());
HeaderComponent = __decorate([
    core_1.Component({
        selector: 'header',
        templateUrl: './header.component.html',
        styleUrls: ['./header.component.less']
    }),
    __metadata("design:paramtypes", [])
], HeaderComponent);
exports.HeaderComponent = HeaderComponent;
var HeaderLeftComponent = (function () {
    function HeaderLeftComponent() {
    }
    return HeaderLeftComponent;
}());
HeaderLeftComponent = __decorate([
    core_1.Component({
        selector: 'header-left',
        template: "\n    <div class=\"navbar-header bg-black\">\n        <ng-content></ng-content>\n    </div>\n  "
    }),
    __metadata("design:paramtypes", [])
], HeaderLeftComponent);
exports.HeaderLeftComponent = HeaderLeftComponent;
var HeaderRightComponent = (function () {
    function HeaderRightComponent() {
        this.dropDown = false;
    }
    HeaderRightComponent.prototype.toggleDropDown = function () {
        this.dropDown = !this.dropDown;
    };
    return HeaderRightComponent;
}());
HeaderRightComponent = __decorate([
    core_1.Component({
        selector: 'header-right',
        template: "\n    <div class=\"collapse pos-rlt navbar-collapse box-shadow\" [ngClass]=\"{'show':dropDown}\">\n        <ng-content></ng-content>\n    </div>\n  "
    }),
    __metadata("design:paramtypes", [])
], HeaderRightComponent);
exports.HeaderRightComponent = HeaderRightComponent;
