// dolphinng/components/sub-nav-item/sub-nav-item.component.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var SubNavItemComponent = (function () {
    function SubNavItemComponent(elemRef) {
        this.elemRef = elemRef;
    }
    SubNavItemComponent.prototype.ngOnInit = function () {
        var _this = this;
        this.elemRef.nativeElement.addEventListener('click', function (ev) {
            if (!_this.isAsideFolded()) {
                ev.stopPropagation();
            }
        });
    };
    SubNavItemComponent.prototype.isAsideFolded = function () {
        var foldedElem = document.querySelector('.app.app-aside-folded');
        var isAsideFolded = foldedElem ? true : false;
        if (!isAsideFolded) {
            return false;
        }
        var classList = foldedElem.className.split(/\s+/);
        if (classList.indexOf('off-screen') >= 0) {
            var clientWidth = document.body.clientWidth;
            if (clientWidth < 768) {
                return false;
            }
        }
        return true;
    };
    return SubNavItemComponent;
}());
__decorate([
    core_1.Input(),
    __metadata("design:type", String)
], SubNavItemComponent.prototype, "text", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", String)
], SubNavItemComponent.prototype, "link", void 0);
SubNavItemComponent = __decorate([
    core_1.Component({
        selector: 'sub-nav-item',
        templateUrl: './sub-nav-item.component.html',
        styleUrls: ['./sub-nav-item.component.less']
    }),
    __metadata("design:paramtypes", [core_1.ElementRef])
], SubNavItemComponent);
exports.SubNavItemComponent = SubNavItemComponent;
