// dolphinng/components/modal/modal.component.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var ModalComponent = (function () {
    function ModalComponent(elemRef) {
        this.elemRef = elemRef;
        this.visible = false; //是否可见
        // @Output('visible') visibleChange:EventEmitter<any>=new EventEmitter();//是否溢出屏幕
        this.visibleChange = new core_1.EventEmitter();
        this.overflow = false; //是否溢出屏幕
        this.animated = true;
        this.size = "md";
        this.isRender = false; //是否渲染
        this.isShow = false; //是否显示
        this.outClickClose = true; //点击外边关闭
        this.showCloseButton = true;
    }
    ModalComponent.prototype.ngOnInit = function () {
    };
    /**
     * 背景点击处理
     */
    ModalComponent.prototype.bgClickAction = function (ev) {
        var wouldClose = this.outClickClose;
        if (wouldClose) {
            var target = ev.target || ev.srcElement;
            if (target === this.elemRef.nativeElement.querySelector('.modal')) {
                this.visibleChange.emit(false);
            }
        }
    };
    ModalComponent.prototype.close = function () {
        this.visibleChange.emit(false);
    };
    ModalComponent.prototype.ngOnChanges = function (changes) {
        var _this = this;
        var isVisible = changes['visible'].currentValue;
        if (isVisible === true) {
            this.isRender = true;
            this.addClass(document.body, 'modal-open');
            setTimeout(function () {
                _this.isShow = true;
            }, 100);
            this.initBodyStyle();
        }
        else if (isVisible === false) {
            this.isShow = false;
            setTimeout(function () {
                _this.isRender = false;
                _this.removeClass(document.body, 'modal-open');
            }, 500);
        }
    };
    /**
     * 初始化modal-body的高度/限高
     */
    ModalComponent.prototype.initBodyStyle = function () {
        var _this = this;
        var isOverflow = !!this.overflow;
        if (!isOverflow) {
            setTimeout(function () {
                var modalBody = _this.elemRef.nativeElement.querySelector('.modal-body'); //模态框中间部分
                var maxHeight = document.documentElement.clientHeight - 183; //上外边距30+下外边距30+头部51+底部70=181 2像素的调整
                modalBody.style.maxHeight = maxHeight + 'px';
                modalBody.style.overflowY = 'auto';
            });
        }
    };
    /**
     * 为元素添加一个类
     * @param elem
     * @param className
     */
    ModalComponent.prototype.addClass = function (elem, className) {
        var classList = elem.className.split(/\s+/);
        if (classList.indexOf(className) < 0) {
            classList.push(className);
            elem.className = classList.join(' ');
        }
    };
    /**
     * 删除某个类
     * @param elem
     * @param className
     */
    ModalComponent.prototype.removeClass = function (elem, className) {
        var classList = elem.className.split(/\s+/);
        var clsIndex = classList.indexOf(className);
        if (clsIndex >= 0) {
            classList.splice(clsIndex, 1);
            elem.className = classList.join(' ');
        }
    };
    return ModalComponent;
}());
__decorate([
    core_1.Input(),
    __metadata("design:type", Boolean)
], ModalComponent.prototype, "visible", void 0);
__decorate([
    core_1.Output(),
    __metadata("design:type", core_1.EventEmitter)
], ModalComponent.prototype, "visibleChange", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", Boolean)
], ModalComponent.prototype, "overflow", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", Boolean)
], ModalComponent.prototype, "animated", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", String)
], ModalComponent.prototype, "size", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", Boolean)
], ModalComponent.prototype, "showCloseButton", void 0);
ModalComponent = __decorate([
    core_1.Component({
        selector: 'modal',
        templateUrl: './modal.component.html',
        styleUrls: ['./modal.component.less']
    }),
    __metadata("design:paramtypes", [core_1.ElementRef])
], ModalComponent);
exports.ModalComponent = ModalComponent;
var ModalHeaderComponent = (function (_super) {
    __extends(ModalHeaderComponent, _super);
    function ModalHeaderComponent() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return ModalHeaderComponent;
}(ModalComponent));
ModalHeaderComponent = __decorate([
    core_1.Component({
        selector: 'modal-header',
        template: "\n    <div class=\"modal-header\" [ngClass]=\"{'modal-header-reduce':showCloseButton}\">\n        <ng-content></ng-content>\n    </div>\n  ",
        styleUrls: ['./modal.component.less']
    })
], ModalHeaderComponent);
exports.ModalHeaderComponent = ModalHeaderComponent;
var ModalBodyComponent = (function () {
    function ModalBodyComponent() {
    }
    return ModalBodyComponent;
}());
ModalBodyComponent = __decorate([
    core_1.Component({
        selector: 'modal-body',
        template: "\n    <div class=\"modal-body\">\n        <ng-content></ng-content>\n    </div>\n  "
    }),
    __metadata("design:paramtypes", [])
], ModalBodyComponent);
exports.ModalBodyComponent = ModalBodyComponent;
var ModalFooterComponent = (function () {
    function ModalFooterComponent() {
    }
    return ModalFooterComponent;
}());
ModalFooterComponent = __decorate([
    core_1.Component({
        selector: 'modal-footer',
        template: "\n    <div class=\"modal-footer\" >\n        <ng-content></ng-content>\n    </div>\n  "
    }),
    __metadata("design:paramtypes", [])
], ModalFooterComponent);
exports.ModalFooterComponent = ModalFooterComponent;
